lib/doctrine.js

```diff
@@ -450,7 +450,7 @@
                 try {
                     this._tag.type = parseType(this._title, this._last);
                     if (!this._tag.type) {
-                        if (!isParamTitle(this._title) && !isReturnTitle(this._title)) {
+                        if (!isParamTitle(this._title) && !isProperty(this._title) && !isReturnTitle(this._title)) {
```

Comment:
I'm a little confused by this logic (prior to this PR as well) - `isTypeParameterRequired` returns `true` for `param` and `return` tags, but then this section of code allows them to pass through without error.

My expectation would be that, instead of `isTypeParameterRequired` indicating that param, property, and return require types, it would return `false` and instead `isAllowedType` would correctly say that they have optional types.

Does that make sense?

Reply:
Hi @tmcw  and @tibdex , it would be great to have this MR in....is there anything I can do to help out? @tmcw 's comment seems to make sense, I can submit a separate PR with those changes if that would help.